express-notes-api/index.js

```diff
@@ -0,0 +1,151 @@
+const express = require('express');
+const fs = require('fs');
+const app = express();
+const JSONparser = express.json();
+
+app.use(JSONparser);
+
+app.get('/api/notes', (req, res) => {
+  const dataArr = [];
+
+  fs.readFile('data.json', 'utf8', (err, data) => {
+    if (err) throw err;
+    const dataObj = JSON.parse(data);
+    const notes = dataObj.notes;
+    for (const key in notes) {
+      dataArr.push(notes[key]);
+    }
+    res.status(200).json(dataArr);
+  });
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = parseInt(req.params.id, 10);
+  if (isNaN(id) || id < 0) {
+    res.status(400).json({ error: 'ID must be a positive interger!' });
+    return;
+  }
+
+  fs.readFile('data.json', 'utf8', (err, data) => {
+    if (err) throw err;
```

Comment:
Instead of throwing (which will crash your whole server), add these:
```js
if (err) {
  console.error(err); // so you can debug errors
  res.status(500).json({
    error: 'an unexpected error occurred'
  }); // to let the client know that the server is not working properly
  return; // to prevent the server from attempting to respond twice.
}
```